weave

```diff
@@ -1393,51 +1425,17 @@ launch_router() {
         --http-addr $HTTP_ADDR \
         --status-addr $STATUS_ADDR \
         --resolv-conf "/var/run/weave/etc/$RESOLV_CONF_BASE" \
-        "$@")
+        $PROXY_ARGS)
     wait_for_status $CONTAINER_NAME http_call $HTTP_ADDR
+    # Wait for Weave Proxy:
+    [ -n "$NO_PROXY" ] || wait_for_status $CONTAINER_NAME http_call_unix $CONTAINER_NAME status.sock
```

Comment:
> I can't see a reason to start the proxy after the main http listener.

OK, I can move it to the beginning of the `main` and see how that goes (which is what I initially had, but wasn't convenient for `signals.SignalHandlerLoop(common.Log, proxy, router)`).